Common/Scheduling/TimeRules.cs

```diff
@@ -215,5 +229,29 @@ private static IEnumerable<DateTime> EveryIntervalIterator(IEnumerable<DateTime>
                 }
             }
         }
+
+        /// <summary>
+        /// Check whether the data resolution is compatible with a interval defined by a <see cref="ITimeRule"/>. 
+        /// </summary>
+        /// <param name="name"><see cref="ITimeRule"/> name</param>
+        /// <param name="interval">The interval defined by a <see cref="ITimeRule"/></param>
+        /// <param name="security">The security with the data subscriptions. If not defined, will use all the available securities</param>
+        public void ValidateInterval(string name, TimeSpan interval, Security security = null)
+        {
+            var resolution = (security?.Subscriptions ?? _securities.SelectMany(x => x.Value.Subscriptions))
```

Comment:
Please update to include the universe resolution as well. Will this work with universe only algorithms?

Not sure "GetHighestResolution()" is the right answer here either. If I have a 1-minute level subscription to schedule events but using the scheduled events to trigger trades on daily data, its the same issue.

Reply:
**TL;DR;** I think we should be checking both, the `SecurityManager` and then the `SubscriptionManager`.

While this particular method shouldn't be called for universe-only algorithms, you raise a good point. I think this should using the `SubscriptionManager`. IIRC, the `SecurityManager` is guaranteed to be updated with subscriptions added to a security _after_ that security has been added, but the `SubscriptionManager` has access to the `DataManager` via its `IAlgorithmSubscriptionManager` member which is capable of providing a direct snapshot of whatever subscriptions are in the data feed since the data feeds us the `DataManager` for subscription storage.... which... since it's the data feed's storage, I think we'll need to use both, since the subscriptions aren't guaranteed to be added _directly_ after adding a security.

Another concern here is that this is an `O(n)` solution. We could prevent the `O(n)` search by maintaining the value in an appropriate place (like the `DataManager`). In fact, the highest resolution should always be the FF resolution, excluding tick.